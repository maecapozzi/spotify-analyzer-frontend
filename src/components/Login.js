import React from 'react'
import PropTypes from 'prop-types'
import LoginButton from './LoginButton'
import Header from './Header'

import injectSheet from 'react-jss'

const Login = ({ classes, url }) => (
  <div>
    <Header string={'Log In'} />
    <h3 className={classes.h3}>Please log in to get a better glimpse of that song you can't get out of your head.</h3>
    <LoginButton url={url} />
  </div>
)

Login.propTypes = {
  url: PropTypes.string
}

const styles = {
  h3: {
    textAlign: 'center',
    color: '#5A6575',
    fontFamily: 'Roboto'
  }
}

export default injectSheet(styles)(Login)
