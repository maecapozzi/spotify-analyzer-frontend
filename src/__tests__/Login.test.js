import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Login from '../components/Login'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const LoginProps = {
    url: 'http://localhost:3001'
  }

  const component = shallow(<Login {...LoginProps} />)
  expect(component).toMatchSnapshot()
})
