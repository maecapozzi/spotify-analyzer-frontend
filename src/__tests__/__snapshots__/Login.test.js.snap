// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Login)
    url="http://localhost:3001"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "h3": "Login-h3-0-1",
      },
      "url": "http://localhost:3001",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "h3": "Login-h3-0-1",
        },
        "url": "http://localhost:3001",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
